sdk/eventhub/event-hubs/src/eventHubConsumerClient.ts

```diff
@@ -1,209 +1,221 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
+import { EventHubClientOptions, EventHubClient } from "./eventHubClient";
+import { PartitionProcessor } from "./partitionProcessor";
+import { ReceivedEventData } from "./eventData";
+import { InMemoryPartitionManager } from "./inMemoryPartitionManager";
+import { EventProcessor, PartitionManager, CloseReason, PartitionContext } from "./eventProcessor";
+import { GreedyPartitionLoadBalancer } from "./partitionLoadBalancer";
 
-import { AbortSignalLike } from "@azure/abort-controller";
-import { isTokenCredential, TokenCredential } from "@azure/core-amqp";
 import {
-  EventHubClient,
-  EventHubClientOptions,
-  GetPartitionIdsOptions,
-  GetPropertiesOptions,
-  GetPartitionPropertiesOptions
-} from "./eventHubClient";
-import { OnError, OnMessage } from "./eventHubReceiver";
-import { EventPosition } from "./eventPosition";
-import { EventHubProperties, PartitionProperties } from "./managementClient";
-import { ReceiveHandler } from "./receiveHandler";
-import { EventHubConsumer } from "./receiver";
+  HostAndTokenCredential,
+  EventHubConnectionString,
+  SubscriptionOptions,
+  Subscription
+} from "./eventHubConsumerClientModels";
+
+export type OnReceivedEvents = (
+  receivedEvents: ReceivedEventData[],
+  context: PartitionContext
+) => Promise<void>;
 
-/**
- * @class
- * The client is the main point of interaction with Azure Event Hubs service.
- * It offers connection to a specific Event Hub within the Event Hubs namespace along with
- * operations for receiving event data and inspecting the connected Event Hub.
- *
- * There are multiple ways to create an `EventHubConsumerClient`
- * - Use the connection string from the SAS policy created for your Event Hub instance.
- * - Use the connection string from the SAS policy created for your Event Hub namespace,
- * and the name of the Event Hub instance
- * - Use the fully qualified domain name of your Event Hub namespace like `<yournamespace>.servicebus.windows.net`,
- * and a credentials object.
- *
- */
 export class EventHubConsumerClient {
-  private _client: EventHubClient;
-  private _consumerGroupName: string;
-  private _consumersMap: Map<string, EventHubConsumer>;
+  private _eventHubClient: EventHubClient;
 
-  /**
-   * @property
-   * @readonly
-   * The name of the Event Hub instance for which this client is created.
-   */
-  get eventHubName(): string {
-    return this._client.eventHubName;
+  constructor(
+    connectionInfo: EventHubConnectionString | HostAndTokenCredential,
+    private consumerGroupName: string,
```

Comment:
One pattern we follow in this SDK is to create private properties that begin with an underscore, and if we want it to be public we create a public getter for it.

In this case, `consumerGroupName` would be stored in `_consumerGroupName`. This is to make it clearer for our non-typescript users that they shouldn't access this field.

Reply:
Done.